src/main/scala/org/hammerlab/guacamole/commands/VAFHistogram.scala

```diff
@@ -179,30 +178,38 @@ object VAFHistogram {
       } else {
         // Print histograms to standard out
         for {
-          histogram <- variantAlleleHistograms
-          sortedHist = histogram.toSeq.sortBy(_._1)
-          (bin, numLoci) <- sortedHist
+          (sampleId, histogram) <- variantAlleleHistograms
+          (bin, numLoci) <- histogram
         } {
-          println(histogramToString(bin, numLoci))
+          println(histogramEntryString(bin, numLoci))
         }
       }
 
       if (args.cluster) {
-        val numClusters = args.numClusters
-        variantLoci.foreach(buildMixtureModel(_, numClusters))
-      }
+        val variantsPerSample =
+          variantLoci
+            .keyBy(_.sampleId)
+            .splitByKey(numVariantsPerSample)
```

Comment:
OK @arahuja, I took a new approach here that retains the mixture-per-sample behavior.

I wrote [a `RDD.splitByKey` function in magic-rdds](https://github.com/hammerlab/magic-rdds/blob/f7309c181589d37af9baae03627fbf085b208091/src/main/scala/org/hammerlab/magic/rdd/SplitByKeyRDD.scala#L7-L16) which will incur one shuffle in order to then pass a separate RDD to the mixture-API for each key (in this case sample ID).

Let me know if that assuages your concerns.